code42_connector.py

```diff
@@ -154,11 +166,90 @@ def _handle_list_high_risk_employees(self, param, action_result):
         action_result.update_summary({"total_count": total_count})
         return action_result.set_status(phantom.APP_SUCCESS)
 
+    """ALERTS ACTIONS"""
+
+    @action_handler_for("get_alert_details")
+    def _handle_get_alert_details(self, param, action_result):
+        alert_id = param["alert_id"]
+        response = self._client.alerts.get_details([alert_id])
+        alert = response.data["alerts"][0]
+        action_result.add_data(alert)
+        action_result.update_summary(
+            {"username": alert["actor"], "user_id": alert["actorId"]}
+        )
+        return action_result.set_status(phantom.APP_SUCCESS)
+
+    @action_handler_for("search_alerts")
+    def _handle_search_alerts(self, param, action_result):
+        username = param.get("username")
+        start_date = param.get("start_date")
+        end_date = param.get("end_date")
+        alert_state = param.get("alert_state")
+
+        if username is None and start_date is None and end_date is None and alert_state is None:
+            return action_result.set_status(
+                phantom.APP_ERROR, "Must supply a search term."
+            )
+
+        if not self._validate_date_range(start_date, end_date):
+            return action_result.set_status(
+                phantom.APP_ERROR,
+                "Start Date and End Date are both required to search by " "date range.",
+            )
+        try:
+            query = self._build_alerts_query(
+                username, start_date, end_date, alert_state
+            )
+        except ValueError as exception:
+            return action_result.set_status(
+                phantom.APP_ERROR,
+                f"Start Date and End Date must be in format YYYY-mm-dd: {exception}",
+            )
+
+        response = self._client.alerts.search(query)
+        action_result.add_data(response.data)
+        action_result.update_summary({"total_count": response["totalCount"]})
+        return action_result.set_status(phantom.APP_SUCCESS)
+
+    @action_handler_for("set_alert_state")
+    def _handle_set_alert_state(self, param, action_result):
+        alert_id = param["alert_id"]
+        alert_state = param["alert_state"]
+        note = param.get("note")
+        response = self._client.alerts.update_state(alert_state, [alert_id], note=note)
+        action_result.add_data(response.data)
+        status_message = f"State of alert {alert_id} was updated to {alert_state}"
+        action_result.update_summary({"alert_id": alert_id})
+        return action_result.set_status(phantom.APP_SUCCESS, status_message)
+
     def finalize(self):
         # Save the state, this data is saved across actions and app upgrades
         self.save_state(self._state)
         return phantom.APP_SUCCESS
 
+    def _validate_date_range(self, start_date, end_date):
+        return (not start_date and not end_date) or (
```

Comment:
It took me quite a while for some reason to understand what was going on here, lots of really simpler logic gates....

Maybe just a comment like `# Requires both args if present` or something